src/daemon/delayed_shutdown_timer.cpp

```diff
@@ -55,13 +55,22 @@ mp::DelayedShutdownTimer::~DelayedShutdownTimer()
 {
     if (shutdown_timer.isActive())
     {
-        if (ssh_session)
+        try
         {
-            // exit_code() is here to make sure the command finishes before continuing in the dtor
-            ssh_session->exec("wall The system shutdown has been cancelled").exit_code();
+            if (ssh_session)
+            {
+                // exit_code() is here to make sure the command finishes before continuing in the dtor
+                ssh_session->exec("wall The system shutdown has been cancelled").exit_code();
+            }
+            mpl::log(mpl::Level::info, virtual_machine->vm_name, fmt::format("Cancelling delayed shutdown"));
+            virtual_machine->state = VirtualMachine::State::running;
+        }
+        catch (const std::exception& e)
```

Comment:
Yup! You almost got me there  :smirk: I was about to complain and then realized this is a dtor, so a boundary where no exception should pass. Good call.